### ISSUE TITLE
Early Stopping in Prompt Tuning

### ISSUE THREAD

**Issue by alex-jw-brooks** [caikit/caikit-nlp#24]
# Background
Currently, we have no early stopping added to the prompt tuning module, but we should! This issue covers adding it.

### Acceptance Criteria
- [ ] Prompt tuning can be run with early stopping
- [ ] Tests added on tiny models validating early stopping via stubbed behaviors or with (fast) deterministic examples

**Comment by chinghuachen** [comment]
This is related to https://github.ibm.com/ai-foundation/watson-fm-stack-tracker/issues/302

**Comment by chinghuachen** [comment]
WCA for Red Hat Ansible Lightspeed has a need for this feature, with the clarification that the tests are performed on a validation test set that is distinct from the training data.
